src/Facebook/FacebookClient.php

```diff
@@ -88,43 +88,33 @@ class FacebookClient
     /**
      * Instantiates a new FacebookClient object.
      *
-     * @param FacebookHttpClientInterface|null $httpClientHandler
-     * @param boolean                          $enableBeta
+     * @param HttpClient|null $httpClient
+     * @param boolean         $enableBeta
      */
-    public function __construct(FacebookHttpClientInterface $httpClientHandler = null, $enableBeta = false)
+    public function __construct(HttpClient $httpClient = null, $enableBeta = false)
     {
-        $this->httpClientHandler = $httpClientHandler ?: $this->detectHttpClientHandler();
+        $this->httpClient = $httpClient ?: HttpClientDiscovery::find();
         $this->enableBetaMode = $enableBeta;
     }
 
     /**
      * Sets the HTTP client handler.
      *
-     * @param FacebookHttpClientInterface $httpClientHandler
+     * @param HttpClient $httpClient
      */
-    public function setHttpClientHandler(FacebookHttpClientInterface $httpClientHandler)
+    public function setHttpClient(HttpClient $httpClient)
```

Comment:
The setter is good for those who use the Facebook PHP SDK via some kind of wrapper lib that doesn't give them access to the constructor injection. There's really no harm in keeping it. :)